tests/PHPStan/Analyser/nsrt/composer-array-bug.php

```diff
@@ -19,22 +19,22 @@ public function doFoo(): void
 			foreach ($this->config['authors'] as $key => $author) {
 				if (!is_array($author)) {
 					$this->errors[] = 'authors.'.$key.' : should be an array, '.gettype($author).' given';
-					assertType("mixed", $this->config['authors']);
```

Comment:
That's not related to the PR, but since we have `!empty($this->config['authors'])` ; shouldn't we already know it's 
```
assertType("mixed~(0|0.0|'0'|''|false|null)",
```
or I misunderstand something ?

Reply:
yeah, it should be more precise.

I guess its something similar to https://github.com/phpstan/phpstan/issues/13312